CompositeAggregationDataExtractorTests#testExtractionGivenCancelHalfWay

**Build scan**: https://gradle-enterprise.elastic.co/s/de34dnnwbe3rg

**Repro line**:
```
./gradlew ':x-pack:plugin:ml:test' --tests "org.elasticsearch.xpack.ml.datafeed.extractor.aggregation.CompositeAggregationDataExtractorTests.testExtractionGivenCancelHalfWay" -Dtests.seed=805366D5B1339577 -Dtests.security.manager=true -Dtests.locale=ar-BH -Dtests.timezone=America/Cayenne -Druntime.java=11
```

**Reproduces locally?**: yes

```
org.elasticsearch.xpack.ml.datafeed.extractor.aggregation.CompositeAggregationDataExtractorTests > testExtractionGivenCancelHalfWay FAILED
    java.lang.AssertionError: 
    Expected: <6L>
         but: was <0L>
        at __randomizedtesting.SeedInfo.seed([805366D5B1339577:792D399757C2AC22]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ml.datafeed.extractor.aggregation.CompositeAggregationDataExtractorTests.testExtractionGivenCancelHalfWay(CompositeAggregationDataExtractorTests.java:336)
```

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/datafeed/extractor/aggregation/CompositeAggregationDataExtractorTests.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/datafeed/extractor/aggregation/CompositeAggregationDataExtractorTests.java
@@ -277,7 +277,7 @@ public class CompositeAggregationDataExtractorTests extends ESTestCase {
                     timestamp,
                     "time_bucket",
                     3,
-                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 1000)), createAvg("responsetime", 32.0)),
+                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 999)), createAvg("responsetime", 32.0)),
                     Collections.singletonList(Tuple.tuple("airline", "c"))
                 )
             );
@@ -304,7 +304,7 @@ public class CompositeAggregationDataExtractorTests extends ESTestCase {
                     timestamp,
                     "time_bucket",
                     3,
-                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 1000)), createAvg("responsetime", 32.0)),
+                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 999)), createAvg("responsetime", 32.0)),
                     Collections.singletonList(Tuple.tuple("airline", "c"))
                 )
             );
@@ -316,7 +316,7 @@ public class CompositeAggregationDataExtractorTests extends ESTestCase {
                     timestamp,
                     "time_bucket",
                     3,
-                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 1000)),
+                    Arrays.asList(createMax("time", randomLongBetween(timestamp, timestamp + 999)),
                         createAvg("responsetime", 32.0)),
                     Collections.singletonList(Tuple.tuple("airline", "c"))
                 )

